collector-sqs/src/main/java/zipkin/collector/sqs/SQSCollector.java

```diff
@@ -172,32 +181,44 @@ public Builder parallelism(int parallelism) {
       if (processors == null) return;
 
       for (SQSSpanProcessor processor : processors) {
-        try {
-          processor.close();
-        } catch (IOException ioe) {
-          logger.log(Level.WARNING, "Processor failed to close cleanly", ioe);
-        }
+        processor.close();
       }
     }
   }
 
-  private static final class LazyAmazonSQSAsync extends LazyCloseable<AmazonSQSAsync> {
+  private static final class LazyAmazonSQSClient extends LazyCloseable<AmazonSQS> {
 
     final AWSCredentialsProvider credentialsProvider;
+    final int waitTimeSeconds;
 
-    LazyAmazonSQSAsync(AWSCredentialsProvider credentialsProvider) {
-      this.credentialsProvider = credentialsProvider;
+    LazyAmazonSQSClient(Builder builder) {
+      this.credentialsProvider = builder.credentialsProvider;
+      this.waitTimeSeconds = builder.waitTimeSeconds;
     }
 
-    @Override protected AmazonSQSAsync compute() {
-      return new AmazonSQSAsyncClient(credentialsProvider);
+    @Override protected AmazonSQS compute() {
+      return new AmazonSQSClient(credentialsProvider);
     }
 
     @Override public void close() {
       AmazonSQS client = maybeNull();
       if (client == null) return;
 
-      client.shutdown();
+      // Close the SQS client after processors have finished consuming last message.  This relies
+      // on the processor being marked closed, which will stop it from consuming after the next
+      // request finishes.  Since the next request may take up to 'waitTimeSeconds' this value
+      // can be used to delay the client shutdown.
+      final Timer timer = new Timer("sqsCollectorCollector");
```

Comment:
Any suggestions here are most welcome :)  I'm not a huge fan of the Java Timer but it was quick to do and hopefully understandable.